DS1307.h

```diff
@@ -50,21 +51,29 @@ class DS1307 {
 
   public:
     void begin();
+#ifndef Arduino_h
     void begin(uint16_t SDA ,uint16_t SCL);
+#endif
     void startClock(void);
     void stopClock(void);
     void setTime(void);
     void getTime(void);
     void fillByHMS(uint8_t _hour, uint8_t _minute, uint8_t _second);
     void fillByYMD(uint16_t _year, uint8_t _month, uint8_t _day);
     void fillDayOfWeek(uint8_t _dow);
+    void setRamAddress(uint8_t _addr, uint8_t _value);
+    void setRam();
+    uint8_t getRamAddress(uint8_t _addr);
+    void getRam();
+    bool isStarted();
     uint8_t second;
     uint8_t minute;
     uint8_t hour;
     uint8_t dayOfWeek;// day of week, 1 = Monday
     uint8_t dayOfMonth;
     uint8_t month;
     uint16_t year;
+    uint8_t ram[56];
```

Comment:
This blocks 56 bytes of precious memory, even if the user doesn't plan to manipulate the RAM directly. What about setRam and getRam accepting this as a parameter instead of an internal variable? In this way, only the users requiring it need to allocate the 56 bytes beforehand.